packages/charts/chart-utilities/src/PlotlySchemaConverter.ts

```diff
@@ -164,68 +153,92 @@ export const mapFluentChart = (input: any): OutputChartType => {
+          validateSeriesData(trace, false);
         }
-      case 'heatmap':
-        return { isValid: true, type: 'heatmap' };
-      case 'sankey':
-        return { isValid: true, type: 'sankey' };
-      case 'indicator':
-      case 'gauge':
-        if (validSchema.data?.[0]?.mode?.includes('gauge') || validSchema.data?.[0]?.type === 'gauge') {
-          return { isValid: true, type: 'gauge' };
+        if (trace.type === 'contour') {
+          throw new Error(`Unsupported chart - type :${trace.type}`);
         }
-        return {
-          isValid: false,
-          errorMessage: `Unsupported chart - type: ${validSchema.data[0]?.type}, mode: ${validSchema.data[0]?.mode}`,
-        };
-      case 'histogram':
-        return validateSeriesData(validSchema, 'verticalbar', false);
-      case 'scatter':
-        if (validSchema.data[0]?.mode === 'markers' && !isNumberArray(validSchema.data[0].y!)) {
-          return {
-            isValid: false,
-            errorMessage: `Unsupported chart - type :${validSchema.data[0]?.type}, mode: ${validSchema.data[0]?.mode}
-           , yAxisType: String or Date`,
-          };
+        if (trace.type === 'bar') {
+          if (trace.orientation === 'h' && trace.base !== undefined) {
+            throw new Error('Unsupported chart type: Gantt');
+          } else if (trace.orientation === 'h' && isNumberArray(trace.x)) {
+            validateSeriesData(trace, false);
+          } else {
+            validateSeriesData(trace, true);
+          }
         }
-        const isAreaChart = validSchema.data.some(
-          (series: PlotData) => series.fill === 'tonexty' || series.fill === 'tozeroy',
-        );
-        return validateDatapointsInternal(validSchema, isAreaChart);
-      case 'contour':
-        return {
-          isValid: false,
-          errorMessage: `Unsupported chart - type :${validSchema.data[0]?.type}`,
-        };
-      default:
-        const xValues = (validSchema.data[0] as PlotData).x;
-        const yValues = (validSchema.data[0] as PlotData).y;
-        if (xValues && yValues && xValues.length > 0 && yValues.length > 0) {
-          return validateDatapointsInternal(validSchema, false);
+        if (trace.type === 'scatter' || isLineData(trace as Partial<PlotData>)) {
+          const scatterTrace = trace as Partial<PlotData>;
+          if (scatterTrace.mode === 'markers' && !isNumberArray(scatterTrace.x)) {
+            throw new Error(
+              `Unsupported chart - type :${scatterTrace.type}, mode: ${scatterTrace.mode}, xAxisType: String or Date`,
+            );
+          } else {
+            validateSeriesData(scatterTrace, true);
+          }
         }
-        return {
-          isValid: false,
-          errorMessage: `Unsupported chart - type :${validSchema.data[0]?.type}}`,
-        };
+
+        return true;
+      } catch (error) {
+        errorMessage = `data[${index}]: ${error}`;
+        return false;
```

Comment:
instead of throwing error and catching you can just return on finding the first valid trace.

Reply:
Or define the filter function separately

Reply:
defined the `getValidDataIndices` function to filter the Plotly data